lib/elixir/lib/exception.ex

```diff
@@ -240,7 +240,7 @@ defmodule Exception do
   defp format_exit_reason(:noconnection), do: "no connection"
 
   defp format_exit_reason(:noproc) do
-    "no process"
+    "no process: the process is dead or no process is associated with the given name, possibly because its application isn't started"
```

Comment:
It needs to be "or" since this error can also be raised by non-atom references. I like the use of currently though because it reflects it may be ephemeral. Maybe: "the process is not alive or no process currently associated with the given name, possibly because its application isn't started"